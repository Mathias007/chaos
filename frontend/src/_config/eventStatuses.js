const eventStatuses = {
    articles: {
        ARTICLE_SUCCESSFULLY_LOADED: "ARTICLE_SUCCESSFULLY_LOADED",
        ARTICLE_NOT_FOUND: "ARTICLE_NOT_FOUND",
        ARTS_LIST_LOADED: "ARTS_LIST_LOADED",
        ARTS_LIST_NOT_FOUND: "ARTS_LIST_NOT_FOUND",
        ARTICLE_ADDED: "ARTICLE_ADDED",
        ARTICLE_ADDING_FAILED: "ARTICLE_ADDING_FAILED",
        ARTICLE_SUCCESFULLY_EDITED: "ARTICLE_SUCCESFULLY_EDITED",
        ARTICLE_EDITING_FAILED: "ARTICLE_EDITING_FAILED",
        ARTICLE_SUCCESFULLY_DELETED: "ARTICLE_SUCCESFULLY_DELETED",
        ARTICLE_DELETING_FAILED: "ARTICLE_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    auth: {
        PASSWORD_TOO_SHORT: "PASSWORD_TOO_SHORT",
        PASSWORD_INCORRECT: "PASSWORD_INCORRECT",
        PASSWORDS_NOT_MATCH: "PASSWORDS_NOT_MATCH",
        USER_LOADED: "USER_LOADED",
        LOGIN_SUCCESSFUL: "LOGIN_SUCCESSFUL",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        LOGIN_FAILED: "LOGIN_FAILED",
        LOGOUT_SUCCESSFUL: "LOGOUT_SUCCESSFUL",
        RESET_STATUS: "RESET_STATUS"
    },
    news: {
        NEWS_SUCCESSFULLY_LOADED: "NEWS_SUCCESSFULLY_LOADED",
        NEWS_NOT_FOUND: "NEWS_NOT_FOUND",
        NEWS_LIST_LOADED: "NEWS_LIST_LOADED",
        NEWS_LIST_NOT_FOUND: "NEWS_LIST_NOT_FOUND",
        NEWS_ADDED: "NEWS_ADDED",
        NEWS_ADDING_FAILED: "NEWS_ADDING_FAILED",
        NEWS_SUCCESFULLY_EDITED: "NEWS_SUCCESFULLY_EDITED",
        NEWS_EDITING_FAILED: "NEWS_EDITING_FAILED",
        NEWS_SUCCESFULLY_DELETED: "NEWS_SUCCESFULLY_DELETED",
        NEWS_DELETING_FAILED: "NEWS_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    knights: {
        KAM_BUILDINGS_LOADED: "KAM_BUILDINGS_LOADED",
        KAM_BUILDINGS_NOT_FOUND: "KAM_BUILDINGS_NOT_FOUND",
        KAM_UNITS_LOADED: "KAM_UNITS_LOADED",
        KAM_UNITS_NOT_FOUND: "KAM_UNITS_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    tzar: {
        TZAR_NATIONS_LOADED: "TZAR_NATIONS_LOADED",
        TZAR_NATIONS_NOT_FOUND: "TZAR_NATIONS_NOT_FOUND",
        TZAR_UNITS_LOADED: "TZAR_UNITS_LOADED",
        TZAR_UNITS_NOT_FOUND: "TZAR_UNITS_NOT_FOUND",
        TZAR_SPELLS_LOADED: "TZAR_SPELLS_LOADED",
        TZAR_SPELLS_NOT_FOUND: "TZAR_SPELLS_NOT_FOUND",
        TZAR_TECHNOLOGIES_LOADED: "TZAR_TECHNOLOGIES_LOADED",
        TZAR_TECHNOLOGIES_NOT_FOUND: "TZAR_TECHNOLOGIES_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    users: {
        USER_SUCCESSFULLY_LOADED: "USER_SUCCESSFULLY_LOADED",
        USER_NOT_FOUND: "USER_NOT_FOUND",
        USERS_LIST_LOADED: "USERS_LIST_LOADED",
        USERS_LIST_NOT_FOUND: "USERS_LIST_NOT_FOUND",
        USER_ADDED: "USER_ADDED",
        USER_ADDING_FAILED: "USER_ADDING_FAILED",
        USER_SUCCESFULLY_EDITED: "USER_SUCCESFULLY_EDITED",
        USER_EDITING_FAILED: "USER_EDITING_FAILED",
        USER_SUCCESFULLY_DELETED: "USER_SUCCESFULLY_DELETED",
        USER_DELETING_FAILED: "USER_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    }
};

export default eventStatuses;
