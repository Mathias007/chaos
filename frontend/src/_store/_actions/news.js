import {
    getAllNews,
    getSingleNews,
    addNews,
    editNews,
    deleteNews
} from "./services/NewsService";

import eventStatuses from "../../_config/eventStatuses";

const {
    NEWS_SUCCESSFULLY_LOADED,
    NEWS_NOT_FOUND,
    NEWS_LIST_LOADED,
    NEWS_ADDED,
    AUTHENTICATION_ERROR,
    NEWS_ADDING_FAILED,
    NEWS_SUCCESFULLY_EDITED,
    NEWS_EDITING_FAILED,
    NEWS_SUCCESFULLY_DELETED,
    NEWS_DELETING_FAILED
} = eventStatuses.news;

export const showProperNews = id => {
    return (dispatch, getState) => {
        const dispatchSingleNewsLoaded = function(news) {
            dispatch({
                type: NEWS_SUCCESSFULLY_LOADED,
                data: news
            });
        };

        const dispatchNewsAuthError = function(response) {
            dispatch({ type: AUTHENTICATION_ERROR, data: response.data });
            throw response.data;
        };

        const dispatchNewsNotFound = function(response) {
            dispatch({ type: NEWS_NOT_FOUND, data: response.data });
            throw response.data;
        };

        return getSingleNews(
            id,
            dispatchSingleNewsLoaded,
            dispatchNewsAuthError,
            dispatchNewsNotFound
        );
    };
};

export const showNewsList = () => {
    return (dispatch, getState) => {
        const dispatchNewsListLoaded = function(news) {
            dispatch({
                type: NEWS_LIST_LOADED,
                data: news
            });
        };

        return getAllNews(dispatchNewsListLoaded);
    };
};

export const addSingleNews = (title, content, author, date, category) => {
    return (dispatch, getState) => {
        const dispatchNewsAdded = function(response) {
            dispatch({
                type: NEWS_ADDED,
                data: response.data
            });
            return response.data;
        };

        const dispatchNewsAuthError = function(response) {
            dispatch({ type: AUTHENTICATION_ERROR, data: response.data });
            throw response.data;
        };

        const dispatchNewsAddingFailed = function(response) {
            dispatch({ type: NEWS_ADDING_FAILED, data: response.data });
            throw response.data;
        };

        return addNews(
            title,
            content,
            author,
            date,
            category,
            dispatchNewsAdded,
            dispatchNewsAuthError,
            dispatchNewsAddingFailed
        );
    };
};

export const editSelectedNews = (
    id,
    title,
    content,
    author,
    lastModified,
    category
) => {
    return (dispatch, getState) => {
        const dispatchNewsEdited = function(response) {
            dispatch({
                type: NEWS_SUCCESFULLY_EDITED,
                data: response.data
            });
            return response.data;
        };

        const dispatchNewsAuthError = function(response) {
            dispatch({ type: AUTHENTICATION_ERROR, data: response.data });
            throw response.data;
        };

        const dispatchNewsEditingFailed = function(response) {
            dispatch({ type: NEWS_EDITING_FAILED, data: response.data });
            throw response.data;
        };

        return editNews(
            id,
            title,
            content,
            author,
            lastModified,
            category,
            dispatchNewsEdited,
            dispatchNewsAuthError,
            dispatchNewsEditingFailed
        );
    };
};

export const deleteSelectedNews = id => {
    return (dispatch, getState) => {
        const dispatchNewsDeleted = function(response) {
            dispatch({
                type: NEWS_SUCCESFULLY_DELETED,
                data: response.data
            });
            return response.data;
        };

        const dispatchNewsAuthError = function(response) {
            dispatch({ type: AUTHENTICATION_ERROR, data: response.data });
            throw response.data;
        };

        const dispatchNewsDeletingFailed = function(response) {
            dispatch({ type: NEWS_DELETING_FAILED, data: response.data });
            throw response.data;
        };

        return deleteNews(
            id,
            dispatchNewsDeleted,
            dispatchNewsAuthError,
            dispatchNewsDeletingFailed
        );
    };
};
