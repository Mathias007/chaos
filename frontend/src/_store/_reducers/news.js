import eventStatuses from "../../_config/eventStatuses";

const {
    NEWS_SUCCESSFULLY_LOADED,
    NEWS_NOT_FOUND,
    NEWS_LIST_LOADED,
    NEWS_ADDED,
    AUTHENTICATION_ERROR,
    NEWS_ADDING_FAILED,
    NEWS_SUCCESFULLY_EDITED,
    NEWS_EDITING_FAILED,
    NEWS_SUCCESFULLY_DELETED,
    NEWS_DELETING_FAILED
} = eventStatuses.news;

const initialState = {
    news: [],
    properNews: {},
    errorMessage: ""
};

export default function news(state = initialState, action) {
    switch (action.type) {
        case NEWS_SUCCESSFULLY_LOADED:
            console.log(action.data.news);
            return {
                ...state,
                ...action.data,
                properNews: action.data.news
            };

        case NEWS_NOT_FOUND:
            console.log(action.data.message);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case NEWS_LIST_LOADED:
            console.log(action.data.news);
            return {
                ...state,
                ...action.data,
                news: action.data.news
            };

        case NEWS_ADDED:
            console.log(action.data.message);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case AUTHENTICATION_ERROR:
            console.log(action.data.message);
            return {
                ...state,
                ...action.data,
                errorMessage: "Something went wrong."
            };

        case NEWS_ADDING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case NEWS_SUCCESFULLY_EDITED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case NEWS_EDITING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case NEWS_SUCCESFULLY_DELETED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case NEWS_DELETING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        default:
            return state;
    }
}
