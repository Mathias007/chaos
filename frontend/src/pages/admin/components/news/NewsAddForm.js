import React, { Component } from "react";
import { connect } from "react-redux";

import { news } from "../../../../_store/_actions";
import linksPaths from "../../../../_config/linksPaths";
import serverStatuses from "../../../../_config/serverStatuses";
import styles from "../../../../styles/styles";

import SingleFormElement from "../../../components/universal/SingleFormElement";
import ButtonComponent from "../../../components/universal/ButtonComponent";
import ErrorMessageComponent from "../../../components/universal/ErrorMessageComponent";
import Success from "../../../components/errors/Success";

import { Form, Layout } from "antd";
const { Content } = Layout;

const { NEWS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class NewsAddForm extends Component {
    handleSubmit = (values) => {
        console.log("Received values of form: ", values);
        this.props.addSingleNews(
            values.title,
            values.content,
            values.author,
            values.date,
            values.category
        );
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={NEWS.MAIN}
                    description="Czy chcesz kontynuować pracę?"
                />
            );
        else {
            return (
                <Content style={styles.content}>
                    <Form onFinish={this.handleSubmit} id="add-news-form">
                        <SingleFormElement
                            fieldName="category"
                            inputIcon="folder-open"
                            label="Kategoria wpisu"
                            message="Określ kategorię wpisu!"
                            placeholder="Określ kategorię wpisu"
                            required
                            tooltip="Kategorie systematyzują wpisy pod kątem merytorycznym."
                        />

                        <SingleFormElement
                            fieldName="title"
                            inputIcon="flag"
                            label="Tytuł wpisu"
                            message="Podaj tytuł wpisu!"
                            placeholder="Nadaj wpisowi tytuł"
                            required
                            tooltip="Tytuł wpisu, wyświetlany jako nagłówek w aktualnościach i na stronie danego wpisu."
                        />

                        <SingleFormElement
                            fieldName="content"
                            fieldType="text-area"
                            label="Zawartość wpisu"
                            message="Umieść zawartość nowego wpisu!"
                            placeholder="Daj ponieść się ekspresji..."
                            required
                            rows={8}
                        />

                        <SingleFormElement
                            fieldName="date"
                            fieldType="date"
                            label="Data dodania lub utworzenia (pole nieobowiązkowe)"
                            inputIcon="calendar"
                            placeholder="Data utworzenia (domyślnie wygenerowana zostanie aktualna)"
                            required={false}
                        />

                        <SingleFormElement
                            fieldName="author"
                            initialValue={this.props.name}
                            inputIcon="crown"
                            label="Autor artykułu"
                            message="Podaj autora wpisu!"
                            placeholder="Podaj autora wpisu"
                            required
                            tooltip="Wpisz nazwę użytkownika, który jest autorem wpisu. Domyślnie jest nim zalogowany użytkownik."
                        />

                        <ButtonComponent
                            htmlType="submit"
                            icon="file-add"
                            text="Dodaj wpis"
                            type="primary"
                        />
                        <ErrorMessageComponent
                            status={this.props.status}
                            errorMessage={this.props.errorMessage}
                        />
                    </Form>
                </Content>
            );
        }
    }
}

const mapStateToProps = (state) => {
    return {
        errorMessage: state.news.errorMessage,
        name: state.auth.name,
        status: state.news.status,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        addSingleNews: (title, content, author, date, category) => {
            return dispatch(
                news.addSingleNews(title, content, author, date, category)
            );
        },
        cleanServerStatus: () => {
            return dispatch(news.cleanServerStatus());
        },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewsAddForm);
