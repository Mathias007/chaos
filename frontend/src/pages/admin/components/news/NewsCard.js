import React, { Component } from "react";
import { connect } from "react-redux";

import { news } from "../../../_store/_actions";
import linksPaths from "../../../_config/linksPaths";
import styles from "../../../styles/styles";

import {
    generateOptions,
    generateComments,
    generateAvatar,
    generateMetaDescription,
    generateNewsCardTitle
} from "./AdminDataGenerators";

import { Card } from "antd";
const { Meta } = Card;

const { GENERAL, NEWS } = linksPaths;

const captions = {
    added: "Dodano:",
    author: "Autor:",
    backText: "Powr√≥t",
    comments: "Komentarze:",
    commentsCount: "brak",
    options: "Opcje"
};

class NewsCard extends Component {
    render() {
        const {
            _id,
            title,
            category,
            content,
            date,
            author
        } = this.props.properNews;

        return (
            <Card
                type="inner"
                style={styles.card}
                title={generateNewsCardTitle(
                    GENERAL.INDEX,
                    captions.backText,
                    title
                )}
                extra={
                    this.props.isAuthenticated ? (
                        <div>
                            {generateOptions(
                                NEWS.EDIT,
                                NEWS.REMOVE,
                                _id,
                                captions.options
                            )}
                        </div>
                    ) : null
                }
                actions={[
                    generateComments(captions.comments, captions.commentsCount)
                ]}
            >
                <Meta
                    avatar={generateAvatar(category)}
                    description={generateMetaDescription(
                        date,
                        captions.added,
                        author,
                        captions.author
                    )}
                />
                <p>{content} </p>
            </Card>
        );
    }
}

const mapStateToProps = state => {
    return {
        isAuthenticated: state.auth.isAuthenticated,
        errorMessage: state.news.errorMessage,
        properNews: state.news.properNews
    };
};

const mapDispatchToProps = dispatch => {
    return {
        showProperNews: id => {
            return dispatch(news.showProperNews(id));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewsCard);
