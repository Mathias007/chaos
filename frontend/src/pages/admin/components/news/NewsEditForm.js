import React, { Component } from "react";
import { connect } from "react-redux";

import { news } from "../../../_store/_actions";
import linksPaths from "../../../_config/linksPaths";
import serverStatuses from "../../../_config/serverStatuses";
import styles from "../../../styles/styles";

import SingleFormElement from "../../components/SingleFormElement";
import ButtonComponent from "../../components/ButtonComponent";
import ErrorMessageComponent from "../../components/ErrorMessageComponent";
import Success from "../../errors/Success";

import { Form, Layout } from "antd";
const { Content } = Layout;

const { NEWS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class NewsEditForm extends Component {
    handleSubmit = e => {
        e.preventDefault();

        let modificationDate = new Date();
        const { validateFields } = this.props.form;

        validateFields((err, values) => {
            if (!err) {
                console.log("Received values of form: ", values);
                this.props.editSelectedNews(
                    this.props.idParam,
                    values.title,
                    values.content,
                    values.author,
                    values.category,
                    modificationDate
                );
            }
        });
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={NEWS.MAIN}
                    description="Czy chcesz kontynuować pracę?"
                />
            );
        else {
            const { getFieldDecorator } = this.props.form;
            const { properNews } = this.props;
            return (
                <Content style={styles.content}>
                    <Form onSubmit={this.handleSubmit} id="edit-news-form">
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="category"
                            initialValue={properNews.category}
                            inputIcon="key"
                            label="Kategoria wpisu"
                            message="Określ kategorię wpisu!"
                            placeholder="Określ kategorię wpisu"
                            required
                            tooltip="Kategorie systematyzują wpisy pod kątem merytorycznym."
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="title"
                            initialValue={properNews.title}
                            inputIcon="flag"
                            label="Tytuł wpisu"
                            message="Wpisz tytuł wpisu!"
                            placeholder="Nadaj wpisowi tytuł"
                            required
                            tooltip="Tytuł wpisu, wyświetlany jako nagłówek w aktualnościach i na stronie danego wpisu."
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="content"
                            fieldType="text-area"
                            initialValue={properNews.content}
                            label="Zawartość wpisu"
                            message="Umieść zawartość nowego wpisu!"
                            placeholder="Daj ponieść się ekspresji..."
                            required
                            rows={8}
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            label="Autor wpisu"
                            fieldName="author"
                            initialValue={properNews.author}
                            inputIcon="crown"
                            message="Podaj autora wpisu!"
                            placeholder="Podaj autora wpisu"
                            required
                            tooltip="Wpisz nazwę użytkownika, który jest autorem wpisu. Domyślnie jest nim zalogowany użytkownik."
                        />

                        <ButtonComponent
                            composition="double"
                            cancelLink={NEWS.MAIN}
                            cancelText="Zrezygnuj"
                            htmlType="submit"
                            icon="edit"
                            text="Edytuj wpis"
                            type="primary"
                        />
                        <ErrorMessageComponent
                            status={this.props.status}
                            errorMessage={this.props.errorMessage}
                        />
                    </Form>
                </Content>
            );
        }
    }
}

const mapStateToProps = state => {
    return {
        errorMessage: state.news.errorMessage,
        properNews: state.news.properNews,
        status: state.news.status
    };
};

const mapDispatchToProps = dispatch => {
    return {
        showProperNews: id => {
            return dispatch(news.showProperNews(id));
        },
        editSelectedNews: (
            id,
            title,
            content,
            author,
            category,
            modificationDate
        ) => {
            return dispatch(
                news.editSelectedNews(
                    id,
                    title,
                    content,
                    author,
                    category,
                    modificationDate
                )
            );
        },
        cleanServerStatus: () => {
            return dispatch(news.cleanServerStatus());
        }
    };
};

NewsEditForm = connect(mapStateToProps, mapDispatchToProps)(NewsEditForm);

export default Form.create()(NewsEditForm);
