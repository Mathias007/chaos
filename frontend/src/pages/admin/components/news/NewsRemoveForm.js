import React, { Component } from "react";
import { connect } from "react-redux";

import { news } from "../../../../_store/_actions";
import linksPaths from "../../../../_config/linksPaths";
import serverStatuses from "../../../../_config/serverStatuses";
import styles from "../../../../styles/styles";

import ButtonComponent from "../../../components/universal/ButtonComponent";
import ErrorMessageComponent from "../../../components/universal/ErrorMessageComponent";
import Success from "../../../components/errors/Success";

import { Layout } from "antd";
const { Content } = Layout;

const { NEWS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class NewsRemoveForm extends Component {
    handleDeletingSubmit = e => {
        e.preventDefault();
        this.props.deleteSelectedNews(this.props.idParam);
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    oneButton
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={NEWS.MAIN}
                    description="Możesz wrócić do listy artykułów"
                />
            );
        else {
            const { properNews } = this.props;
            return (
                <Content style={styles.content}>
                    <p>
                        Czy na pewno chcesz usunąć artykuł o tytule{" "}
                        <strong>{properNews.title}</strong>? Tej operacji nie
                        będzie można cofnąć.
                    </p>

                    <ButtonComponent
                        composition="double"
                        cancelLink={NEWS.MAIN}
                        cancelText="Zrezygnuj"
                        htmlType="submit"
                        icon="delete"
                        onClick={this.handleDeletingSubmit}
                        text="Usuń wpis"
                        type="primary"
                    />
                    <ErrorMessageComponent
                        status={this.props.status}
                        errorMessage={this.props.errorMessage}
                    />
                </Content>
            );
        }
    }
}

const mapStateToProps = state => {
    return {
        errorMessage: state.news.errorMessage,
        properNews: state.news.properNews,
        status: state.news.status
    };
};

const mapDispatchToProps = dispatch => {
    return {
        deleteSelectedNews: id => {
            return dispatch(news.deleteSelectedNews(id));
        },
        cleanServerStatus: () => {
            return dispatch(news.cleanServerStatus());
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewsRemoveForm);
