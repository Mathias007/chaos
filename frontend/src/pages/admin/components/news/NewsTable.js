import React, { Component } from "react";
import { connect } from "react-redux";

import { news } from "../../../../_store/_actions";
import styles from "../../../../styles/styles";

import { newsColumnsStructure } from "../_helpers/articlesColumnsStructure";

import { Layout, Table } from "antd";
const { Content } = Layout;

class NewsTable extends Component {
    state = {
        tableColumns: [],
        tableData: []
    };

    componentDidMount() {
        this.props.showNewsList();
        console.log(this.props.news);
    }

    componentDidUpdate(prevProps) {
        if (this.props.news.length !== prevProps.news.length) {
            this.props.showNewsList();
        }
    }

    renderNews() {
        let { news } = this.props;

        if (news) {
            return news.map((singleNews, index) => {
                const {
                    _id,
                    title,
                    content,
                    author,
                    date,
                    category
                } = singleNews;

                return {
                    number: index,
                    key: _id,
                    title,
                    content,
                    author,
                    date,
                    category,
                    options: _id
                };
            });
        }
    }

    render() {
        let { tableColumns, tableData } = this.state;

        tableData = this.renderNews();
        tableColumns = newsColumnsStructure;

        return (
            <Content style={styles.content}>
                <Table dataSource={tableData} columns={tableColumns} />
            </Content>
        );
    }
}

const mapStateToProps = state => {
    return {
        news: state.news.news
    };
};

const mapDispatchToProps = dispatch => {
    return {
        showNewsList: () => dispatch(news.showNewsList())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewsTable);
