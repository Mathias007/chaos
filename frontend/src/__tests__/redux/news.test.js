import eventStatuses from "../../_config/eventStatuses";
import news from "../../_store/_reducers/news";

const {
    NEWS_SUCCESSFULLY_LOADED,
    NEWS_NOT_FOUND,
    NEWS_LIST_LOADED,
    NEWS_LIST_NOT_FOUND,
    NEWS_ADDED,
    NEWS_ADDING_FAILED,
    NEWS_SUCCESFULLY_EDITED,
    NEWS_EDITING_FAILED,
    NEWS_SUCCESFULLY_DELETED,
    NEWS_DELETING_FAILED,
    AUTHENTICATION_ERROR
} = eventStatuses.news;

const initialState = {
    news: [],
    properNews: {},
    errorMessage: ""
};

describe("test news reducer", () => {
    const newsList = {
        message: "Pozytywna wiadomość z serwera!",
        news: [
            {
                _id: "1111-2222-3333-4444",
                category: "test-news",
                title: "Wpis testowy",
                content: "Zawartość wpisu testowego",
                author: "tester",
                date: "01-01-2020"
            }
        ]
    };

    const singleNews = {
        message: "Okuratna wiadomość z serwera!",
        singleNews: {
            _id: "1111-2222-3333-4444",
            category: "test-news",
            title: "Wpis testowy",
            content: "Zawartość wpisu testowego",
            author: "tester",
            date: "01-01-2020"
        }
    };

    const serverMessage = { message: "Ważna wiadomość z serwera!" };

    it("should return the initial state", () => {
        expect(news(undefined, {})).toEqual(initialState);
    });

    it("should load news list", () => {
        const action = { type: NEWS_LIST_LOADED, data: newsList };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about news list's not finding", () => {
        const action = { type: NEWS_LIST_NOT_FOUND, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should load a single news", () => {
        const action = {
            type: NEWS_SUCCESSFULLY_LOADED,
            data: singleNews
        };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about single news' not finding", () => {
        const action = { type: NEWS_NOT_FOUND, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about adding news to database", () => {
        const action = { type: NEWS_ADDED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about news' adding fail", () => {
        const action = { type: NEWS_ADDING_FAILED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about modifying news in database", () => {
        const action = { type: NEWS_SUCCESFULLY_EDITED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about news's modifying fail", () => {
        const action = { type: NEWS_EDITING_FAILED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about removing news from database", () => {
        const action = { type: NEWS_SUCCESFULLY_DELETED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about news' removing fail", () => {
        const action = { type: NEWS_DELETING_FAILED, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });

    it("should get message about a problem with authentication", () => {
        const action = { type: AUTHENTICATION_ERROR, data: serverMessage };
        expect(news(initialState, action)).toMatchSnapshot();
    });
});
