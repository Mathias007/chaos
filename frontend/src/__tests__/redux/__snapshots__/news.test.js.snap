// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test news reducer should get message about a problem with authentication 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about adding news to database 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about modifying news in database 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about news list's not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": null,
}
`;

exports[`test news reducer should get message about news' adding fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about news' removing fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about news's modifying fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about removing news from database 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": 200,
}
`;

exports[`test news reducer should get message about single news' not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {},
  "status": null,
}
`;

exports[`test news reducer should load a single news 1`] = `
Object {
  "errorMessage": "Okuratna wiadomość z serwera!",
  "message": "Okuratna wiadomość z serwera!",
  "news": Array [],
  "properNews": Object {
    "_id": "1111-2222-3333-4444",
    "author": "tester",
    "category": "test-news",
    "content": "Zawartość wpisu testowego",
    "date": "01-01-2020",
    "title": "Wpis testowy",
  },
  "singleNews": Object {
    "_id": "1111-2222-3333-4444",
    "author": "tester",
    "category": "test-news",
    "content": "Zawartość wpisu testowego",
    "date": "01-01-2020",
    "title": "Wpis testowy",
  },
  "status": null,
}
`;

exports[`test news reducer should load news list 1`] = `
Object {
  "errorMessage": "Pozytywna wiadomość z serwera!",
  "message": "Pozytywna wiadomość z serwera!",
  "news": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "author": "tester",
      "category": "test-news",
      "content": "Zawartość wpisu testowego",
      "date": "01-01-2020",
      "title": "Wpis testowy",
    },
  ],
  "properNews": Object {},
  "status": null,
}
`;
